python/cuml/solvers/qn.pyx

```diff
@@ -310,14 +312,24 @@ class QN(Base,
 
         self._num_classes = len(cp.unique(y_m))
 
+        cdef uintptr_t sample_weight_ptr = 0
+        if sample_weight is not None:
+            sample_weight, _, _, _ = \
+                input_to_cuml_array(sample_weight,
+                                    check_dtype=self.dtype,
+                                    convert_to_dtype=(self.dtype
+                                                      if convert_dtype
+                                                      else None))
+            sample_weight_ptr = sample_weight.ptr
+
         self.loss_type = self._get_loss_int(self.loss)
         if self.loss_type != 2 and self._num_classes > 2:
             raise ValueError("Only softmax (multinomial) loss supports more"
                              "than 2 classes.")
 
         if self.loss_type == 2 and self._num_classes <= 2:
-            raise ValueError("Only softmax (multinomial) loss supports more"
-                             "than 2 classes.")
+            raise ValueError("Softmax (multinomial) loss should only be"
+                             "used with more than 2 classes.")
```

Comment:
The prior wording of the message was more clear, no? i.e. this is informing the user that they have 2 classes and are trying to use softmax for that. This wording would be better on the error of line 327

Reply:
Personally, I find that the wording is a bit strange for this. I think that the constant here is the number of classes that the user probably has no choice over during his work. Then we're trying to prevent the working from using an inappropriate loss function. A better wording might be "Two classes or less cannot be trained with softmax (multinomial)". What do you think?

Reply:
I like that new wording indeed, sounds much more clear :)